### DISCUSSION TITLE
How does LNS in CP-SAT avoid local optimal?

### DISCUSSION THREAD

**Discussion by C8504** [google/or-tools#3374] [CP-SAT questions] [answered]
Hi Experts，

Could I ask a question about how does LNS in CP-SAT avoid local optimal? Especially in below LNS subsolver:

rnd_var_lns
rnd_cst_lns
graph_var_lns
graph_cst_lns

Thanks.
John

**Comment by lperron** [comment, marked as answer]
It does not depend on the neighborhood. By relaxing enough, or the right variables, you can expect to include an improving solution in you reoptimization. 

It is not guaranteed to succeed though.
